pandas-stubs/io/formats/style.pyi

```diff
@@ -1,97 +1,267 @@
-jinja2 = ...
-has_mpl: bool = ...
-no_mpl_message: str = ...
+from typing import (
+    Any,
+    Callable,
+    Hashable,
+    Sequence,
+    TypeVar,
+)
 
-class Styler:
-    loader = ...
-    env = ...
-    template = ...
-    ctx = ...
-    data = ...
-    index = ...
-    columns = ...
-    uuid = ...
-    table_styles = ...
-    caption = ...
-    precision = ...
-    table_attributes = ...
-    hidden_index: bool = ...
-    hidden_columns = ...
-    cell_ids = ...
-    na_rep = ...
+from pandas.core.frame import DataFrame
+from pandas.core.series import Series
+
+from pandas._typing import (
+    Axis,
+    FilePath,
+    IndexLabel,
+    Level,
+    Scalar,
+    WriteBuffer,
+)
+
+from pandas.io.formats.style_render import (
+    CSSProperties,
+    CSSStyles,
+    ExtFormatter,
+    StylerRenderer,
+    Subset,
+)
+
+_StylerT = TypeVar("_StylerT", bound=Styler)
+
+class Styler(StylerRenderer):
     def __init__(
         self,
-        data,
-        precision=...,
-        table_styles=...,
-        uuid=...,
-        caption=...,
-        table_attributes=...,
-        cell_ids=...,
+        data: DataFrame | Series,
+        precision: int | None = ...,
+        table_styles: CSSStyles | None = ...,
+        uuid: str | None = ...,
+        caption: str | tuple | None = ...,
+        table_attributes: str | None = ...,
+        cell_ids: bool = ...,
         na_rep: str | None = ...,
-    ): ...
+        uuid_len: int = ...,
+        decimal: str | None = ...,
+        thousands: str | None = ...,
+        escape: str | None = ...,
+        formatter: ExtFormatter | None = ...,
+    ) -> None: ...
+    # def render(self,sparse_index: bool | None = ...,sparse_columns: bool | None = ...,**kwargs,) -> str: ...
+    def set_tooltips(
+        self,
+        ttips: DataFrame,
+        props: CSSProperties | None = ...,
+        css_class: str | None = ...,
+    ) -> Styler: ...
     def to_excel(
         self,
         excel_writer,
         sheet_name: str = ...,
         na_rep: str = ...,
-        float_format=...,
-        columns=...,
-        header: bool = ...,
+        float_format: str | None = ...,
+        columns: Sequence[Hashable] | None = ...,
+        header: Sequence[Hashable] | bool = ...,
         index: bool = ...,
-        index_label=...,
+        index_label: IndexLabel | None = ...,
         startrow: int = ...,
         startcol: int = ...,
-        engine=...,
+        engine: str | None = ...,
         merge_cells: bool = ...,
-        encoding=...,
+        encoding: str | None = ...,
         inf_rep: str = ...,
         verbose: bool = ...,
-        freeze_panes=...,
+        freeze_panes: tuple[int, int] | None = ...,
```

Comment:
need to add `storage_options`

Reply:
There is a comment about this - storage options is not actually in the function declatation in pandas.

Reply:
So can you create a pandas issue indicating the error in the docs?